src/graphics/program-lib/standard.js

```diff
@@ -138,34 +138,39 @@ pc.programlib.standard = {
         transformUv1: { n: "transformVS", f: _oldChunkTransformUv1 }
     },
 
-    generateKey: function (device, options) {
-        var prop, props = [];
-        var key = "standard";
-        var light;
-        for (prop in options) {
-            if (options.hasOwnProperty(prop)) {
-                if (prop === "chunks") {
-                    for (var p in options[prop]) {
-                        if (options[prop].hasOwnProperty(p)) {
-                            props.push(p + options.chunks[p]);
-                        }
-                    }
-                } else {
-                    if (options[prop] !== undefined) props.push(prop);
-                }
+    generateKey: function (options) {
+        if (!this.props) {
+            this.props = [];
```

Comment:
You shouldn't really use ```this``` here - it works, but it's a bit hard to read, because this module isn't a 'class', it's an object with a number of functions in it. I probably would have declared ```props``` as a property via JSON notation, namely: ```props: [],```